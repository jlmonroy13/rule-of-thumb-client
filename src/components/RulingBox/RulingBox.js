import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon';

const RulingBox = ({ title, description, wikiUrl }) => (
  <div className="ruling-box">
    <span className="text-small">What's your opinion on</span>
    <h1 className="ruling-box__title">{title}</h1>
    <p className="ruling-box__description">{description}</p>
    <div className="mb-6">
      <Icon name="wikipedia" className="mr-1" />
      <a href={wikiUrl} target="_blank" rel="noopener noreferrer" className="ruling-box__link">More information</a>
    </div>
    <span className="font-weight-bold text-larger">What's Your Verdict?</span>
    <div className="ruling-box__buttons">
      <button className="ruling-box__button">
        <Icon name="thumb" className="ruling-box__icon ruling-box__icon--up" />
      </button>
      <button className="ruling-box__button ruling-box__button--yellow">
        <Icon name="thumb" className="ruling-box__icon" />
      </button>
    </div>
  </div>
);

RulingBox.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  wikiUrl: PropTypes.string.isRequired,
};

export default RulingBox;
