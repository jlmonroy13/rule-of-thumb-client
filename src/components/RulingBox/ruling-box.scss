.ruling-box {
  background-color: rgba($black, 0.3);
  backdrop-filter: blur(16px);
  -webkit-backdrop-filter: blur(10px);
  padding: 30px 60px 30px 30px;
  color: $white;
  position: relative;
  bottom: 80px;
  max-width: 500px;

  &__title {
    @include font-size(60);
    line-height: 60px;
    font-weight: 400;
    margin-bottom: 25px;
  }

  &__description {
    @include font-size(21);
    line-height: 24px;
    font-weight: 300;
  }

  &__link {
    @include font-size(14);
    color: $white;
    text-decoration: underline;

    &:hover {
      color: $white;
    }
  }

  &__buttons {
    position: absolute;
    bottom: -80px;
    left: 0;
    width: 100%;
    display: flex;
    justify-content: space-between;
  }

  &__button {
    flex-basis: 0;
    flex-grow: 1;
    background-color: rgba($main-color, 0.9);
    padding: 20px 0;

    &:hover {
      background-color: rgba($main-color, 1);
    }

    &--yellow {
      background-color: rgba($secondary-color, 0.9);

      &:hover {
      background-color: rgba($secondary-color, 1);
    }
    }
  }

  &__icon {
    @include font-size(40);

    &--up {
      transform: rotate(180deg)
    }
  }
}
