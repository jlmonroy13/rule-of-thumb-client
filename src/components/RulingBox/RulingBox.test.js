/**
 *
 * Tests for RulingBox
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import RulingBox from './index';

const props = {
  title: 'Pope Francis?',
  description: 'He is talking tough on clergy sexual abuse, but is he just another papal pervert protector? (thumbs down) or a true pedophile punishing pontiff? (thumbs up)',
  wikiUrl: 'https://en.wikipedia.org/wiki/Catholic_Church_sexual_abuse_cases'
};

describe('<RulingBox />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<RulingBox {...props} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(true);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<RulingBox {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
