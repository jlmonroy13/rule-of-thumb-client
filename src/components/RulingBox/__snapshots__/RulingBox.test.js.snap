// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RulingBox /> Should render and match the snapshot 1`] = `
<div
  class="ruling-box"
>
  <span
    class="text-small"
  >
    What's your opinion on
  </span>
  <h1
    class="ruling-box__title"
  >
    Pope Francis?
  </h1>
  <p
    class="ruling-box__description"
  >
    He is talking tough on clergy sexual abuse, but is he just another papal pervert protector? (thumbs down) or a true pedophile punishing pontiff? (thumbs up)
  </p>
  <div
    class="mb-6"
  >
    <svg
      class="icon icon-wikipedia mr-1"
      data-testid="icon-component"
    >
      <use
        xlink:href="#icon-wikipedia"
      />
    </svg>
    <a
      class="ruling-box__link"
      href="https://en.wikipedia.org/wiki/Catholic_Church_sexual_abuse_cases"
      rel="noopener noreferrer"
      target="_blank"
    >
      More information
    </a>
  </div>
  <span
    class="font-weight-bold text-larger"
  >
    What's Your Verdict?
  </span>
  <div
    class="ruling-box__buttons"
  >
    <button
      class="ruling-box__button"
    >
      <svg
        class="icon icon-thumb ruling-box__icon ruling-box__icon--up"
        data-testid="icon-component"
      >
        <use
          xlink:href="#icon-thumb"
        />
      </svg>
    </button>
    <button
      class="ruling-box__button ruling-box__button--yellow"
    >
      <svg
        class="icon icon-thumb ruling-box__icon"
        data-testid="icon-component"
      >
        <use
          xlink:href="#icon-thumb"
        />
      </svg>
    </button>
  </div>
</div>
`;
